src/main/java/com/fasterxml/jackson/databind/ser/impl/SimpleBeanPropertyFilter.java

```diff
@@ -218,6 +218,8 @@ public void serializeAsField(Object pojo, JsonGenerator jgen,
             writer.serializeAsField(pojo, jgen, provider);
         } else if (!jgen.canOmitFields()) { // since 2.3
             writer.serializeAsOmittedField(pojo, jgen, provider);
+        } else if (writer instanceof AnyGetterWriter) {
```

Comment:
WIP : Contemplating on how should we optimize this check...

Reply:
I tried referring to other writer/serializers and it turns out `instanceof` here check seems sort of reasonable.